lib/ClangImporter/ImportDecl.cpp

```diff
@@ -3618,7 +3618,17 @@ namespace {
             if (functionDecl->isTemplateInstantiation()) {
               return;
             }
+            if (const auto *methodDecl =
+                    dyn_cast<clang::CXXMethodDecl>(functionDecl)) {
+              if (const auto *parent = methodDecl->getParent()) {
+                if (parent->getDescribedClassTemplate() ||
+                    isa<clang::ClassTemplateSpecializationDecl>(parent)) {
+                  return;
+                }
+              }
+            }
```

Comment:
nit: can you add a comment briefly describing why we are foregoing this warning for any functions related to templates?

I also wonder whether there is a more precise way to go about this. What we really want to avoid is giving spurious/non-actionable diagnostics when the _return_ type is dependent on some template instantiation, i.e.,

```c++
functionDecl->getReturnType()->isDependentType()
```

(or perhaps `isInstantiationDependentType()` is the right thing to check here?)

That way, we will still get appropriate warnings for functions like:

```c++
template<typename T>
FRT *ReturnTypeNotDependent(T t);
```